src/google/protobuf/compiler/parser_unittest.cc

```diff
@@ -869,6 +869,22 @@ TEST_F(ParseMessageTest, ReservedNames) {
       "}");
 }
 
+TEST_F(ParseMessageTest, ReservedIdentifiers) {
+  ExpectParsesTo(
+      "edition = \"2023\";\n"
+      "message TestMessage {\n"
+      "  reserved foo, bar;\n"
```

Comment:
Can you add test points that combinations of identifier, string, and field number still work?

Reply:
Whoops, these *won't* work by design

Reply:
I'm not sure I understand. You are not allowed to combine numbers and names currently. Do you mean a test that makes sure this is still disallowed?

Or do you mean a test file that has separate statements, one with numbers and one with names?

Reply:
I wasn't familiar enough with our reserved syntax, disregard :P